remote-processor/RemoteProcessorServer.cpp

```diff
@@ -89,10 +99,19 @@ void CRemoteProcessorServer::stop()
 
     // Cause exiting of the thread
     uint8_t ucData = 0;
-    write(_aiInbandPipe[1], &ucData, sizeof(ucData));
+    while (write(_aiInbandPipe[1], &ucData, sizeof(ucData)) == -1) {
```

Comment:
code is wrong : you don't know if the write will actually write the full number of bytes.